src/js/core/MesosStream.ts

```diff
@@ -1,11 +1,90 @@
+import { Observable, of, timer } from "rxjs";
+import { fromFetch } from "rxjs/fetch";
+import { map, repeatWhen, shareReplay, switchMap } from "rxjs/operators";
 // @ts-ignore
-import { stream } from "@dcos/mesos-client";
-import { timer } from "rxjs";
-import { repeatWhen } from "rxjs/operators";
+import proto from "../mesos-proto";
+
+const eventCoder = proto.lookup("Event");
+
+const utf8Coder = new TextDecoder("utf-8");
```

Comment:
thinking about it, i actually never left the prototyping-mode as the integration-tests still don't seem to run. thus the polishing-mode was not activated yet.

it surely is a `decoder`. but right now i'm wondering whether we can do without `TextDecoder` at all. i'll give that a try and rename in case that idea does not work out.